Documentation/General/emergentArchitectureDesign/emergentArchitectureDesign.tex

```diff
@@ -0,0 +1,180 @@
+
+\section{Design goals}
+The following design goals will be maintained throughout the project:
+\begin{itemize}
+
+\item
+\textbf{Deployability}
+\\
+The system will be developed in such a way that we can always deploy the most current version (Continuous Integration \cite{Duvall}).
+Being able to deploy the system at any time allows us to keep the work required for a release manegable.
+Because the current version should be deployable at anytime, developers are enforced to only alter the system in a non-breaking way.
+Another advantage is that smaller releases\footnote{With smaller we mean releases that touches only a few lines or a few files of the codebase.} imply a lower possibility of introducing bugs.
+
+\item
+\textbf{Portability}
+\\
+The frontend should work on \textit{Evergreen} browsers \footnote{<<The term "evergreen browser" refers to browsers that are automatically upgraded to future versions, rather than being updated by distribution of new versions from the manufacturer, as was the case with older browsers.>> - http://www.techopedia.com/definition/31094/evergreen-browser}.
+The backend should run on both Windows and Linux server environments.
+
+\item
+\textbf{Simplicity}
+\\
+The system will be developed with simplicity in mind.
+Existing libraries and frameworks will be investigated and used if applicable, so we can focus on developing the system rather than the tools required to build the system.
+This will keep our codebase relatively small and limited to the product.
+% By using dependencies we do not reduce the overall complexity, in fact, these dependencies will most likely include unused funcitonality and come with a learning curve for the developers.
+% ^ Not sure what to do with this line, please take into account with review - JW
+
+\item
+\textbf{Object-oriented programming}
+\\
+Object-oriented programming\cite{Wirfs-Brock} will be used to develop the system.
+We will separate the system into subsystems and components (\glspl{Package}, \glspl{Class} and \glspl{Interface}).
+This should enable code reuse and extensibility, and improve the testability of the sytem, because each component can be tested individually\cite{Binder}.
+
+\item
+\textbf{Usability}
+\\
+Users should be able to use Moodcat intuitively.
+% Insert more
+
+\end{itemize}
+
+\chapter{Software architecture views}
+This chapter discusses the architecture of the system.
+The system is first decomposed into smaller subsystems and the dependencies between the subsystems are explained.
+In the second paragraph the relation between the hardware and software of the system is elaborated.
+The third paragraph illustrates the data management of the system.
+
+\section{Subsystem decomposition}
+Moodcat is subdivided into two components: the frontend and the backend service.
```

Comment:
"the frontend and the backend service"
A Front-end website and a Backend processing server (or something like that)